xc/device.ml

```diff
@@ -1819,19 +1820,35 @@ module Dm_Common = struct
     ; fd_map = []
     }
 
-  let vgpu_args_of_nvidia domid vcpus (vgpu:Xenops_interface.Vgpu.nvidia) pci restore device =
+  let vgpu_args_of_nvidia domid vcpus vgpus restore =
     let open Xenops_interface.Vgpu in
+    let virtual_pci_address_compare vgpu1 vgpu2 =
+    match vgpu1, vgpu2 with
+    |{implementation = Nvidia nvi1},  {implementation = Nvidia nvi2} ->
+         begin
+             match nvi1.virtual_pci_address, nvi2.virtual_pci_address with
+             |Some pci1, Some pci2 -> Pervasives.compare pci1.dev pci2.dev
+             |_,_-> 0
+         end
+    |_,_->0 in
+
+    let get_pci_string = function
+      | None -> raise (Xenopsd_error (Internal_error "No PCI address"))
+      | Some pci -> Xcp_pci.string_of_address pci in
+
+    let device_args =
+      List.map (fun x ->
+          match x.implementation with
+          | Nvidia _conf ->
+             "--device=" ^ (Xenops_interface.Pci.string_of_address x.physical_pci_address) ^ ","  ^_conf.config_file ^ "," ^ get_pci_string _conf.virtual_pci_address
```

Comment:
I think it is better to use `Printf.sprintf` to construct these kind of strings. Also avoid overlong lines and break this up.

Reply:
Good idea.